### ISSUE TITLE
Difficult to use the "Encryption Config" screen with VoiceOver on iOS

### ISSUE THREAD

**Issue by personalizedrefrigerator** [laurent22/joplin#11846]
### Operating system

iOS

### Joplin version

3.3.0

### Current behaviour

We've received more feedback on Mastodon -- it's difficult to use the "encryption config" screen on iOS:
> @joplinapp Good afternoon,I signed up this morning for a pro account & now cannot encrypt my notes. I know my master password & have entered it several times. Does anyone know how I can fix this? I've tried uninstalling & reinstalling the app several times, to no avail.  I’m experiencing this issue on my #iPhone15Pro & can’t access the forum, due to not being able to reset that password.
>
> — https://mastodon.hams.social/@KE8UPE/113992396166080165

For reference, here's what I get when using VoiceOver on an encryption config screen with no errors:
<details><summary>VoiceOver transcript</summary>

```
Synchronization: Sync, encryption, proxy
[Double-taps]
Synchronize only over WiFi connection
[Swipes right]
Switch button, synchronize only over WiFi connection, double-tap to toggle setting
[Swipes right]
Encryption Config, button
[Double-taps]
Encryption config. Disable encryption, button.
[Swipes right]
Disabled keys, collapsed, button.
[Double-taps]
Disabled keys, expanded, button.
[Swipes right]
Disabled keys, expanded, button.
[Swipes right]
Disabled keys, expanded, button.
[Swipes left]
Disable encryption, button.
[Swipes left]
Loaded
[Swipes left]
Master password
[Swipes left]
Decrypted items: 56/56
[Swipes left]
Encryption is: Enabled
[Swipes left]
Status
[Swipes left]
https://joplinapp.org/help/apps/sync/e2ee
[Swipes left]
For more information about End-To-End-Encryption (E2EE) and advice on how to enable it, please check the documentation:
[Swipes left]
Encryption config, heading
[Swipes left]
Back, button
```

</details>

There are a few things I noticed:
1. The link is not labeled as a link.
2. The status heading is not labelled as a heading.
3. The "disabled keys" toggle is readable even when there  are no disabled keys.

Additionally, I don't have any decryption-related errors (or secondary keys). Issues may exist related to the "invalid"/"valid" check/cross icon.

<details><summary>On a different iOS device, after opening the encryption config screen</summary>

```
Synchronization: Sync, encryption, proxy
[Double-tap]
Check synchronization configuration, button
[Swipe right]
Synchronization interval
[Swipe left]
Check synchronization configuration, button
[Swipe left]
Connect to Joplin Cloud, button
[Swipe left]
Joplin cloud, down triangle, button, synchronization target, opens dropdown
[Swipe right]
Connect to Joplin Cloud, button
[Swipe right]
Check synchronization configuration, button
[Swipe right]
Synchronization interval
[Swipe right]
5 minutes, down triangle, synchronization interval, opens dropdown
[Swipe right]
Synchronize only over wifi connection
[Swipe right]
Switch button, off, synchronize only over wifi connection, double-tap to toggle setting
[Swipe right]
Encryption config, button
[Double-tap]
Disabled keys, collapsed, button
[Double-tap]
Disabled keys, expanded
[Swipe right]
Master key cb49bb
[Swipe right]
Created 06/12/2023, 16:57
[Swipe right]
Password:
[Swipe right]
Master password
[Swipe right]
Master key d3b298
[Swipe right]
Created: 07/07/2023 21:24
[Swipe right]
Save, button
[Swipe right]
Password:
[Swipe right]
Secure text field, double-tap to edit
[Swipe right]
Cross mark emoji
[Swipe right]
Master key d55202
```

</details>

Above:
- Focus visits the "save" button before visiting the label for the password input.
   - **Note**: Changing [focus order on React Native iOS may be difficult in some cases](https://github.com/react-native-community/discussions-and-proposals/discussions/389) without adding a new dependency or writing native code.
- The "incorrect password" icon is described as "crossmark emoji".
- It isn't immediately clear to me that there are no decryption errors and that everything is working successfully. Changing "Status" to a heading (which can be more easily navigated to) might help with this.

### Expected behaviour

- All items in the "encryption config" screen should have appropriate ARIA `role` information.
- There should not be accessibility issues when entering a password.
